### PR TITLE
#613 - Add the possibility to change date formatting

### PR DESCRIPTION
**Pull request by kisiseldwarf**

### CODE REVIEW

File: totalRP3/Modules/Register/Main/RegisterList.lua
Comment (Meorawr) [inline totalRP3/Modules/Register/Main/RegisterList.lua:1086]: This needs to not be a global function; would recommend renaming it and all usages to `TRP3_API.util.getPreferredDateFormat` for now as that'll likely be what's wanted by Extended.

File: totalRP3/Core/Enums.lua
Comment (Meorawr) [inline totalRP3/Core/Enums.lua:50]: For now I'm tempted to suggest just hardcoding the "date_format" config key in the areas it's being used; longer-term I want to rework the config key nonsense and would rather not have to _also_ factor in this enum,

File: totalRP3/Core/Configuration.lua
Comment (Meorawr) [inline totalRP3/Core/Configuration.lua:391]: I'll need to think some more on the default here. That currently we're using D/M/Y makes this at least consistent, but I feel as if this should actually be "intelligent" and use a proper default based off client locale - but that might confuse US players if we now swapped D/M/Y to M/D/Y for them.

File: totalRP3/Locales/Locale.lua
Comment (Meorawr) [inline totalRP3/Locales/Locale.lua:627]: There's a header section further down the Locale script for adding new localization strings - it should go underneath that section.

**Review by Meorawr — CHANGES_REQUESTED** [review]

File: totalRP3/Core/Enums.lua
Comment (kisiseldwarf) [reply totalRP3/Core/Enums.lua:50]: Oof, yeah it hurts to do that but I understand that my changes would add more areas to look over for your refacto !

File: totalRP3/Core/Configuration.lua
Comment (kisiseldwarf) [reply totalRP3/Core/Configuration.lua:391]: I agree that making it smart would only confuse players on why their settings has changed without their approval

**Review by Meorawr — DISMISSED** [review]
Minus the random whitespace stuff Luacheck complained about it looks fine - we'll think about the default format some more over the weekend and tweak it if necessary.

**Comment by kisiseldwarf** [comment]
Oops. didn't know it would dismiss your review (First time I do a PR on a collaborative project :sweat_smile:

**Review by Solanya — APPROVED** [review]

**Comment by Solanya** [comment]
Thanks, will be sure to credit your help in the changelog 👍

**Comment by kisiseldwarf** [comment]
No problem !
I would love to continue to contribute to this project, if you need anything done let me know

**Pull request merged**
